GeeksCoreLibrary/Components/OrderProcess/Services/OrderProcessesService.cs

```diff
@@ -491,102 +491,119 @@ public async Task<List<PaymentMethodSettingsModel>> GetPaymentMethodsAsync(ulong
         var dataTable = await databaseConnection.GetAsync(query);
         var results = new List<PaymentMethodSettingsModel>();
-        foreach (DataRow dataRow in dataTable.Rows)
+        if (dataTable.Rows.Count == 0)
         {
-            results.Add(await DataRowToPaymentMethodSettingsModelAsync(dataRow));
+            return results;
         }
-
-        // get total amount of order
+        
+        // Get total amount of order.
+        // This is needed to account for minimum and maximum prices.
         var shoppingBaskets = await shoppingBasketsService.GetShoppingBasketsAsync();
         var basketSettings = await shoppingBasketsService.GetSettingsAsync();
         var totalPrice = 0M;
         foreach (var (main, lines) in shoppingBaskets)
         {
             totalPrice += await shoppingBasketsService.GetPriceAsync(main, lines, basketSettings);
         }
-
-        // check if paymentmethods are allowed for the price that the costumer should pay
-        results = results.Where(paymentMethod =>
+        
+        foreach (DataRow dataRow in dataTable.Rows)
         {
+            var paymentMethod = await DataRowToPaymentMethodSettingsModelAsync(dataRow);
+            
             // check if total price is below the minimal
             if (paymentMethod.UseMinimalAmountCheck && totalPrice < paymentMethod.MinimalAmountCheck)
             {
-                return false;
+                continue;
             }
 
             // check if total price is above the maximum
             if (paymentMethod.UseMaximumAmountCheck && totalPrice > paymentMethod.MaximumAmountCheck)
             {
-                return false;
+                continue;
+            }
+            
+            if (loggedInUser != null && paymentMethod.Visibility switch
+                {
+                    OrderProcessFieldVisibilityTypes.Always => false,
+                    OrderProcessFieldVisibilityTypes.WhenNotLoggedIn => loggedInUser.UserId != 0,
+                    OrderProcessFieldVisibilityTypes.WhenLoggedIn => loggedInUser.UserId == 0,
+                    _ => throw new ArgumentOutOfRangeException(nameof(paymentMethod.Visibility), paymentMethod.Visibility.ToString(), null)
+                })
+            {
+                continue;
+            }
+            
+            if (paymentMethod.PaymentMethodLanguageCodes.Count != 0)
+            {
+                var languageCode = await languagesService.GetLanguageCodeAsync();
+                if (!paymentMethod.PaymentMethodLanguageCodes.Contains(languageCode, StringComparer.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
             }
 
-            // amount is within range or checks are disabled
-            return true;
-        }).ToList();
+            var originalPath = httpContextAccessor.HttpContext?.Items[Modules.Templates.Models.Constants.OriginalPathAndQueryStringKey]?.ToString();
```

Comment:
Please define the `originalPath` outside of the `foreach` loop because it doesn't change depending on the paymentmethod.